core/src/lib.rs

```diff
@@ -1,25 +1,40 @@
 use rustls::client::danger::{HandshakeSignatureValid, ServerCertVerified, ServerCertVerifier};
 use rustls::client::{ClientConfig, WebPkiServerVerifier};
-use rustls::crypto::{aws_lc_rs, CryptoProvider};
+use rustls::crypto::CryptoProvider;
 use rustls::pki_types::{CertificateDer, ServerName, UnixTime};
 use rustls::{DigitallySignedStruct, Error as RustlsError, RootCertStore, SignatureScheme};
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
 use std::sync::{Arc, Mutex};
 
+#[cfg(not(target_arch = "wasm32"))]
+use rustls::crypto::aws_lc_rs;
+
+#[cfg(target_arch = "wasm32")]
+use rustls::crypto::ring;
```

Comment:
Looking at [Ring](https://github.com/briansmith/ring) ([crate](https://crates.io/crates/ring) as well), I'm now wondering if we can actually use it.

Reply:
If you read their docs, it feels it's an experiment. I feel their wording isn't clear, but they don't say it's for prod.